### PR TITLE
[cpu] Support tl.load(..., padding="nan")

### LINKED ISSUE #60: Block pointer padding_option="nan" not working
It just pads with zeros instead. (This seems to be the only source of failures in `test_block_pointer.py`.)
 
Repro:
```
import triton
import triton.language as tl
import torch


@triton.jit
def kernel(a_ptr, b_ptr):
    a_block_ptr = tl.make_block_ptr(base=a_ptr,
                                    shape=(32, ),
                                    strides=(1, ),
                                    offsets=(0, ),
                                    block_shape=(64, ),
                                    order=(0, ))
    a = tl.load(a_block_ptr, boundary_check=(0, ), padding_option="nan")
    b = b_ptr + tl.arange(0, 64)
    tl.store(b, a)


a = torch.ones([64])
b = torch.ones([64])
kernel[(1, )](a, b)
print(b)
```

### PR DESCRIPTION
**Pull request by int3**
This makes test_block_pointer.py pass.

Fixes #60.

### CODE REVIEW

File: third_party/cpu/lib/TritonToTritonCPU/ConvertMemoryOps.cpp [hunk 1]
```diff
@@ -126,8 +150,10 @@ struct LoadOpConversion : public MemoryOpConversion<triton::LoadOp> {
     for (auto dim : boundaryChecks) {
       inBounds[dim] = false;
     }
-    auto vecRead = rewriter.create<vector::TransferReadOp>(loc, resTy, memRef,
-                                                           indices, inBounds);
+    Value padding = getPaddingValue(loc, resTy.getElementType(),
```
Comment (int3) [inline third_party/cpu/lib/TritonToTritonCPU/ConvertMemoryOps.cpp:153]: I don't think it matters whether I get the type from the result type or the input type, but not 100% sure...

File: third_party/cpu/lib/TritonToTritonCPU/ConvertMemoryOps.cpp [hunk 2]
```diff
@@ -96,6 +96,30 @@ struct MemoryOpConversion : public OpConversionPattern<OpT> {
...
+    TypedAttr attr;
+    switch (padding.value()) {
+    case triton::PaddingOption::PAD_ZERO:
+      attr = type.isIntOrIndex() ? cast<TypedAttr>(IntegerAttr::get(type, 0))
+                                 : cast<TypedAttr>(FloatAttr::get(type, 0));
+      break;
+    case triton::PaddingOption::PAD_NAN:
+      assert(!type.isIntOrIndex());
+      auto apNaN = llvm::APFloat::getNaN(
+          FloatAttr::get(type, 0).getValue().getSemantics());
```
Comment (int3) [inline third_party/cpu/lib/TritonToTritonCPU/ConvertMemoryOps.cpp:115]: this seems like a rather awkward way to get the NaN value, but it's more or less what RewriteTensorPointer.cpp does

File: third_party/cpu/lib/TritonToTritonCPU/ConvertMemoryOps.cpp [hunk 3]
```diff
@@ -96,6 +96,30 @@ struct MemoryOpConversion : public OpConversionPattern<OpT> {
...
+  getPaddingValue(Location loc, Type type,
+                  const std::optional<triton::PaddingOption> &padding,
+                  ConversionPatternRewriter &rewriter) {
+    if (!padding.has_value())
+      return Value();
+
+    TypedAttr attr;
+    switch (padding.value()) {
+    case triton::PaddingOption::PAD_ZERO:
+      attr = type.isIntOrIndex() ? cast<TypedAttr>(IntegerAttr::get(type, 0))
```
Comment (ienkovich) [inline third_party/cpu/lib/TritonToTritonCPU/ConvertMemoryOps.cpp:109]: You can use `rewriter.getZeroAttr()` here

File: third_party/cpu/lib/TritonToTritonCPU/ConvertMemoryOps.cpp (hunk 2 above)
Comment (ienkovich) [reply third_party/cpu/lib/TritonToTritonCPU/ConvertMemoryOps.cpp:115]: Yeah, there is no good helper in rewriter to build a NaN. For the semantics, I'd use `cast<FloatType>(type).getFloatSemantics()`.

File: third_party/cpu/lib/TritonToTritonCPU/ConvertMemoryOps.cpp (hunk 1 above)
Comment (ienkovich) [reply third_party/cpu/lib/TritonToTritonCPU/ConvertMemoryOps.cpp:153]: The result type usage seems right.

**Review by ienkovich — APPROVED** [review]
Looks good, thanks!

**Comment by int3** [comment]
@minjang I haven't addressed comments yet 😅 also I introduced a small bug which I want to fix... will put up a separate PR I guess

Since I have merge permissions now, could you let me do the merging for my own PRs next time? Thanks!

**Comment by minjang** [comment]
> @minjang I haven't addressed comments yet 😅 also I introduced a small bug which I want to fix... will put up a separate PR I guess
> 
> Since I have merge permissions now, could you let me do the merging for my own PRs next time? Thanks!

Lol. Sorry for that. Yes, I will leave the PRs later on!
